libs/nillion-chain/client/src/client.rs

```diff
@@ -370,6 +369,16 @@ impl NillionChainClient {
             .map_err(|e| anyhow!("signed tx to bytes: {e}"))
     }
 
+    fn calculate_gas_price(gas_needed: u64, gas_price_in_unil: f64) -> u128 {
```

Comment:
Isn't this the same as this?

```
(gas_needed as f64 * gas_price_in_unil).ceil() as u128
```

Reply:
i just tested the above, appears to work exactly the same. 

I have also moved gas adjustment into that function with tests.